packages/core/src/local-workspace/state.ts

```diff
@@ -87,17 +90,15 @@ export const localState = (filePrefix: string): state.State => {
     let elementsData: string[] = []
     let updateDateData: string[] = []
     let pathIndexData: string[] = []
-    let currentFilePaths: string[] = []
     let versions: string[] = []
-    await new Promise<void>(resolve => {
-      glob(`${currentFilePrefix}*${ZIPPED_STATE_EXTENSION}`, (_err: Error | null, files: string[]) => {
-        currentFilePaths = files
-        resolve()
-      })
-    })
+    const currentFilePaths = await glob(`${currentFilePrefix}.*${ZIPPED_STATE_EXTENSION}`)
```

Comment:
Good point about env names with `.` - is it worth just changing this a non-wildcard variant?
```suggestion
    const currentFilePaths = await glob(`${currentFilePrefix}.@(${Object.keys(adapterCreators).join('|')})${ZIPPED_STATE_EXTENSION}`)
```

`adapterCreators` would need to be exported from `core` so we can probably just export only the adapter names. Or we could use `await inMemState.existingServices()`, but the current implementation seems heavier.

Looks like the `clear` [implementation](https://github.com/salto-io/salto/blob/master/packages/core/src/local-workspace/state.ts#L191) is already assuming only services are supported, so I'm not sure we should support the wildcard here. wdyt?

Reply:
added the suggestion in https://github.com/salto-io/salto/pull/1456 after rebasing on top of these changes (I think it also handles some missed backward-compatibility scenarios for rename / clear)